src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -3450,6 +3450,81 @@ public async Task OpenApi_InteractiveSwaggerUI(
             }
         }
 
+        /// <summary>
+        /// Test different loglevel values that are avaliable by deserializing RuntimeConfig with LogLevel value
+        /// and checks if value exists properly inside the deserialized RuntimeConfig.
+        /// </summary>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(Level.Trace, DisplayName = "Validates that log level Trace deserialized correctly")]
+        [DataRow(Level.Debug, DisplayName = "Validates log level Debug deserialized correctly")]
+        [DataRow(Level.Information, DisplayName = "Validates log level Information deserialized correctly")]
+        [DataRow(Level.Warning, DisplayName = "Validates log level Warning deserialized correctly")]
+        [DataRow(Level.Error, DisplayName = "Validates log level Error deserialized correctly")]
+        [DataRow(Level.Critical, DisplayName = "Validates log level Critical deserialized correctly")]
+        [DataRow(Level.None, DisplayName = "Validates log level None deserialized correctly")]
+        [DataRow(null, DisplayName = "Validates log level Null deserialized correctly")]
+        public void TestExistingLogLevels(Level expectedLevel)
+        {
+            RuntimeConfig configWithCustomLogLevel = InitRuntimeWithLogLevel(expectedLevel);
+
+            string configWithCustomLogLevelJson = configWithCustomLogLevel.ToJson();
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(configWithCustomLogLevelJson, out RuntimeConfig deserializedRuntimeConfig));
+
+            Assert.AreEqual(expectedLevel, deserializedRuntimeConfig.Runtime.LogLevel.Value);
+        }
+
+        /// <summary>
+        /// Test different loglevel values that do not exist to ensure that the build fails when they are trying to be set up
+        /// </summary>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(-1, DisplayName = "Validates that a negative log level value, fails to build")]
+        [DataRow(7, DisplayName = "Validates that a positive log level value that does not exist, fails to build")]
+        [DataRow(12, DisplayName = "Validates that a bigger positive log level value that does not exist, fails to build")]
+        public void TestNonExistingLogLevels(Level expectedLevel)
+        {
+            RuntimeConfig configWithCustomLogLevel = InitRuntimeWithLogLevel(expectedLevel);
+
+            // Try should fail and go to catch exception
+            try
+            {
+                string configWithCustomLogLevelJson = configWithCustomLogLevel.ToJson();
+                Assert.Fail();
```

Comment:
How does your `LogLevelOptionsConverterFactory.Write(...)` function behave? Another good test would be similar to this one here, but testing valid values for log level, then seeing how they are written to a json file. 
Additonally, you'd want to test this scenario:
- user creates dab-config.json, but doesn't define a value for, nor include the `log-level` property. When the converter.Write() function is called, does it always write out the `log-level` property/default value to the json file? We don't necessarily want to write properties to files when users didn't explicitly include them.

Reply:
Created new tests called `LogLevelSerialization` and `LogLevelSerializationForNonExistentProperty`